oauth2_http/java/com/google/auth/oauth2/DefaultCredentialsProvider.java

```diff
@@ -211,8 +212,9 @@ private final GoogleCredentials getDefaultCredentialsUnsynchronized(
   }
 
   private void warnAboutProblematicCredentials(GoogleCredentials credentials) {
-    if (credentials instanceof UserCredentials &&
-        ((UserCredentials)credentials).getClientId().equals(CLOUDSDK_CLIENT_ID)) {
+    if (!Boolean.parseBoolean(getEnv(SUPPRESS_GCLOUD_CREDS_WARNING_ENV_VAR))
```

Comment:
Should we make this the last check (i.e. only check the environment variable if we would log the warning?).

Otherwise, this check will be performed more times than necessary (whenever we attempt to load credentials from the well known location).

Reply:
@chingor13 I've reordered the checks. Thanks for the review!